src/components/article/AddArticleForm.jsx

```diff
@@ -0,0 +1,1041 @@
+  const [selectTeam, setSelectTeam] = useState([]);
+
+  const selectedSportId = watch("sportId");
+  const selectedSportIdd = watch("sportIdd");
+  const selectedRoundId = watch("roundId");
+
+  useEffect(() => {
+    if (selectedSportId) {
+      const filtered = roundData?.data?.filter(
+        (round) => round?.sport?._id === selectedSportId
+      );
+      setFilteredRounds(filtered);
+    } else {
+      setFilteredRounds([]);
+    }
+  }, [selectedSportId, roundData, setValue]);
+
+  useEffect(() => {
+    if (selectedSportIdd) {
+      const filteredTeams = allTeamData?.data.filter(
+        (team) => team?.sport?._id === selectedSportIdd
+      );
+      setSelectTeam(filteredTeams || []);
+    }
+  }, [selectedSportIdd, allTeamData, setValue]);
+
+  // Filter games based on selected round
+  useEffect(() => {
+    if (selectedRoundId) {
+      const games = gameData?.data.filter(
+        (game) => game?.round?._id === selectedRoundId
+      );
+      setFilteredGames(games || []);
+    } else {
+      setFilteredGames([]);
+    }
+  }, [selectedRoundId, gameData]);
+
+  ///// Image Uploader
+  const [teamDetails, setTeamDetails] = useState({
+    file: null,
+    htmlFile: null,
+  });
+
+  const [image, setImage] = useState(null);
+  const [htmlFileName, setHtmlFileName] = useState(null);
+
+  const handleImageChange = (e) => {
+    const file = e.target.files[0];
+
+    if (file) {
+      if (!file.type.startsWith("image/")) {
+        alert("Please select a valid image file.");
+        return;
+      }
+      setTeamDetails((prevDetails) => ({
+        ...prevDetails,
+        file: file,
+      }));
+
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setImage(reader.result);
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+  const onHandleHtmlImageChange = (e) => {
+    const file = e.target.files[0];
+
+    if (file) {
+      if (!file.type.startsWith("image/")) {
+        alert("Please select a valid image file.");
+        return;
+      }
+      setTeamDetails((prevDetails) => ({
+        ...prevDetails,
+        htmlFile: file,
+      }));
+
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setHtmlFileName(reader.result);
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+  const navigate = useNavigate();
+  const onhandleSubmit = async (data) => {
+    // e.preventDefault();
+
+    const formData = new FormData();
+    formData.append("title", data?.title);
+    formData.append("content", data?.text);
+    formData.append("addedby", data?.addedby);
+    formData.append("files", teamDetails?.file);
+    formData.append("htmlFiles", teamDetails?.htmlFile);
+    formData.append("game", data?.gameId);
+    formData.append("url", data?.url);
+    formData.append("isActive", true);
+    formData.append("publishDateTime", data?.publishDateTime);
+    formData.append("teamId", data?.teamId);
+    formData.append("articleType", data?.articleType);
```

Comment:
use mapping for these keys like
const dataKeys = ["title", "content", "addedby", "game", "url", "publishDateTime", "teamId", "articleType"];
// create this array in constant file
dataKeys.forEach(key => {
  formData.append(key, data?.[key]);
});